cmd/kubectl-gadget/utils/trace.go

```diff
@@ -290,32 +450,160 @@ RetryLoop:
+	if err != nil {
+		return fmt.Errorf("Error in creating setting up Kubernetes client: %w", err)
+	}
+
 	if customResultsDisplay == nil {
-		if outputMode != "Stream" {
-			panic(fmt.Errorf("OutputMode=%q needs a custom display function", outputMode))
+		// If we arrived here, it means we have at least one successful node, so it
+		// is safe to use trace pointer.
+		if trace.Spec.OutputMode != "Stream" {
+			return fmt.Errorf("OutputMode=%q needs a custom display function", trace.Spec.OutputMode)
+		}
+
+		params := &CommonFlags{
+			OutputMode: config.CommonFlags.OutputMode,
+			Node:       config.CommonFlags.Node,
+			Verbose:    config.CommonFlags.Verbose,
 		}
-		genericStreamsDisplay(contextLogger, client, params, &results, transformLine)
+
+		genericStreamsDisplay(client, params, &traces, transformLine)
 	} else {
-		customResultsDisplay(contextLogger, nodes, &results)
+		nodes, err := client.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			return fmt.Errorf("Error listing nodes: %w", err)
+		}
+
+		customResultsDisplay(nodes, &traces)
+	}
+
+	return nil
+}
+
+// DeleteTrace deletes the traces for the given trace ID using RESTClient.
+func DeleteTrace(traceID string) error {
+	traceRestClient, err := getRestClient()
+	if err != nil {
+		return err
 	}
-	deleteTraces(contextLogger, traceRestClient, traceID)
+
+	deleteTraces(traceRestClient, traceID)
+
+	return nil
+}
+
+// getTraceListFromParameters returns traces associated with the given config.
+func getTraceListFromParameters(config *TraceConfig) ([]gadgetv1alpha1.Trace, error) {
+	namespace := ""
+	if !config.CommonFlags.AllNamespaces {
+		namespace = GetNamespace()
+	}
+
+	// If some of the arguments are empty, they will just be ignored when
+	// searching for corresponding trace.
+	labels := fmt.Sprintf("gadgetName=%v,nodeName=%v,namespace=%v,podName=%v,containerName=%v,outputMode=%v",
+		config.GadgetName, config.CommonFlags.Node, namespace, config.CommonFlags.Podname, config.CommonFlags.Containername, config.TraceOutputMode)
+
+	var listTracesOptions = metav1.ListOptions{
+		LabelSelector: labels,
+		FieldSelector: fields.Everything().String(),
+	}
+
+	traces, err := getTraceListFromOptions(listTracesOptions)
+	if err != nil {
+		return []gadgetv1alpha1.Trace{}, err
+	}
+
+	return traces.Items, nil
+}
+
+// PrintAllTraces prints all traces corresponding to the given config.CommonFlags.
+func PrintAllTraces(config *TraceConfig) error {
+	traces, err := getTraceListFromParameters(config)
+	if err != nil {
+		return err
+	}
+
+	w := tabwriter.NewWriter(os.Stdout, 0, 0, 4, ' ', 0)
+
+	fmt.Fprintln(w, "NAMESPACE\tPOD\tCONTAINER\tTRACEID")
+
+	for _, trace := range traces {
+		filter := trace.Spec.Filter
+
+		id, present := trace.ObjectMeta.Labels[TRACE_ID]
+		if !present {
+			return errors.New("Trace has no ID!")
+		}
+
+		fmt.Fprintf(w, "%v\t%v\t%v\t%v\n",
+			filter.Namespace, filter.Podname, filter.ContainerName, id)
```

Comment:
It is important to remark into the description of the `list` subcommand that we will add to the different gadgets, that this is the filter that the traces are using to run the gadgets and not the trace's information. I don't know if it is just me but I got confused.